fastapi_sqla/sqla.py

```diff
@@ -155,3 +145,36 @@ def get_users(session: fastapi_sqla.Session = Depends()):
             await loop.run_in_executor(None, session.rollback)
 
     return response
+
+
+class SessionDependency:
+    def __init__(self, key: str = _DEFAULT_SESSION_KEY) -> None:
+        self.key = key
+
+    def __call__(self, request: Request) -> SqlaSession:
+        """Yield the sqlalchemy session for that request.
+
+        It is meant to be used as a FastAPI dependency::
+
+            from fastapi import APIRouter, Depends
+            from fastapi_sqla import SqlaSession, SessionDependency
+
+            router = APIRouter()
+
+            @router.get("/users")
+            def get_users(session: SqlaSession = Depends(SessionDependency())):
+                pass
+        """
+        try:
+            return request.scope[f"{_REQUEST_SESSION_KEY}_{self.key}"]
+        except KeyError:
+            logger.exception(
+                f"No session with key '{self.key}' found in request, "
+                "please ensure you've setup fastapi_sqla.",
+                session_key=self.key,
+            )
+            raise
+
+
+default_session_dep = SessionDependency()
```

Comment:
Addresses [this comment](https://github.com/dialoguemd/fastapi-sqla/pull/105#issuecomment-1795317685) for easier overrides